online-shop/src/app/components/presentational/products-list-view/products-list-view.component.ts

```diff
@@ -1,14 +1,16 @@
 import { Component, Input } from '@angular/core';
 import { Product } from 'src/app/modules/shared/types/products.types';
-import { ShoppingCartService } from 'src/app/services/shopping-cart-service';
+import { ShoppingCartService } from 'src/app/modules/shared/shopping-cart/services/shopping-cart-service';
 @Component({
   selector: 'app-products-list-view',
   templateUrl: './products-list-view.component.html',
   styleUrls: ['./products-list-view.component.scss'],
 })
 export class ProductsListViewComponent {
   @Input() products: Product[] | undefined;
-  constructor(private cartService: ShoppingCartService) {} // Inject the ShoppingCartService
+  constructor(private cartService: ShoppingCartService) {
```

Comment:
You should not inject a service and have business logic in the presentational view. Please move the logic to the smart component by emitting an event.

Reply:
Moved the logic to product-list.